frontend/packages/console-shared/src/hooks/useQueryParams.ts

```diff
@@ -0,0 +1,5 @@
+import { useLocation } from 'react-router-dom';
+
+export const useQueryParams = () => {
+  return new URLSearchParams(useLocation().search);
+};
```

Comment:
We want the result to be stable and only change when search has changed.
```suggestion
export const useQueryParams = () => {
  const search = useLocation().search;
  const queryParams = React.useMemo(() => new URLSearchParams(search), [search]);
  return queryParams;
};
```